src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ValidateToken.Internal.cs

```diff
@@ -35,21 +35,34 @@ internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
         }
 
         internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
-            SamlSecurityToken samlToken,
+            SecurityToken securityToken,
             ValidationParameters validationParameters,
             CallContext callContext,
 #pragma warning disable CA1801 // Review unused parameters
             CancellationToken cancellationToken)
 #pragma warning restore CA1801 // Review unused parameters
         {
-            if (samlToken is null)
+            if (securityToken is null)
             {
                 StackFrames.TokenNull ??= new StackFrame(true);
                 return ValidationError.NullParameter(
-                    nameof(samlToken),
+                    nameof(securityToken),
                     StackFrames.TokenNull);
             }
 
+            if (securityToken is not SamlSecurityToken samlToken)
```

Comment:
`securityToken is not SamlSecurityToken samlToken` -> `securityToken is not SamlSecurityToken`

Reply:
If we remove the new variable we have to cast `securityToken` to `SamlSecurityToken` down the line.